app-arch/zip/zip-3.0-r4.ebuild

```diff
@@ -0,0 +1,63 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+inherit toolchain-funcs eutils flag-o-matic
+
+MY_P="${PN}${PV//.}"
+DESCRIPTION="Info ZIP (encryption support)"
+HOMEPAGE="http://www.info-zip.org/"
+SRC_URI="mirror://sourceforge/infozip/${MY_P}.zip"
+
+LICENSE="Info-ZIP"
+SLOT="0"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~m68k ~mips ~ppc ~ppc64 ~s390 ~sparc ~x86 ~x64-cygwin ~amd64-linux ~x86-linux"
+IUSE="bzip2 crypt natspec unicode"
+
+DEPEND="${RDEPEND}"
+RDEPEND="bzip2? ( app-arch/bzip2 )
+	natspec? ( dev-libs/libnatspec )"
+BDEPEND="app-arch/unzip"
+
+S=${WORKDIR}/${MY_P}
+
+src_prepare() {
+	default
+	eapply "${FILESDIR}"/${P}-no-crypt.patch #238398
+	eapply "${FILESDIR}"/${P}-pic.patch
+	eapply "${FILESDIR}"/${P}-exec-stack.patch #122849
+	eapply "${FILESDIR}"/${P}-build.patch #200995
+	eapply "${FILESDIR}"/${P}-zipnote-freeze.patch #322047
+	eapply "${FILESDIR}"/${P}-format-security.patch #512414
+	use natspec && eapply "${FILESDIR}"/${PN}-3.0-natspec.patch #275244
```

Comment:
Please move all patches that are applied unconditionally into an array called `PATCHES`.

Reply:
Done